staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/configmaplock.go

```diff
@@ -56,6 +58,16 @@ func (cml *ConfigMapLock) Get() (*LeaderElectionRecord, error) {
 			return nil, err
 		}
 	}
+	if cml.LeaseLock != nil {
+		leaseLock, err := cml.LeaseLock.Get()
+		if err != nil {
+			return nil, err
+		}
+		// check if the older resource lock is released
```

Comment:
Can you describe the logic behind this in more details - I think I'm not really following this.

Reply:
The main reason here is not to destroy the original interface, only return a resource lock, we want to panic at any time of the lock release. We should check if the earlier lock is expired. (Here should be renew time instead of acquire time, I will fix it tomorrow)